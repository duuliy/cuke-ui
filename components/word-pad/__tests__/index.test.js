import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import WordPad from "../index";

describe("<WordPad/>", () => {
  it("should render a <WordPad/> components", () => {
    const wrapper = render(<WordPad />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render custom style", () => {
    const wrapper = render(
      <WordPad
        width={200}
        height={200}
        strokeColor="#396"
        strokeWidth={3}
        style={{
          border: "1px solid #dcdcdc",
          margin: "10px 0"
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it.skip("should clear canvas when clear props is true", () => {
    const onClearComplete = jest.fn();
    const wrapper = shallow(
      <WordPad clear={true} onClearComplete={onClearComplete} />
    );

    wrapper.setProps({ clear: true }, () => {
      expect(wrapper).toHaveBeenCalled();
    });
  });
});
