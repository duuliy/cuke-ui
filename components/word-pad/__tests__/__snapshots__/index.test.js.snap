// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordPad/> should clear canvas when clear props is true 1`] = `
<canvas
  class="cuke-word-pad"
>
  你的浏览器不支持 canvas
</canvas>
`;

exports[`<WordPad/> should render a <WordPad/> components 1`] = `
<canvas
  class="cuke-word-pad"
>
  你的浏览器不支持 canvas
</canvas>
`;

exports[`<WordPad/> should render custom style 1`] = `
<canvas
  class="cuke-word-pad"
  style="border:1px solid #dcdcdc;margin:10px 0"
>
  你的浏览器不支持 canvas
</canvas>
`;
